pkg/cli/admin/upgrade/status/status.go

```diff
@@ -76,6 +83,23 @@ func (o *options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string
 		o.mockData.nodesPath = strings.Replace(o.mockData.cvPath, cvSuffix, "-node.yaml", 1)
 	}
 
+	kcmdutil.RequireNoArguments(cmd, args)
+
+	cfg, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig = cfg
+
+	routeClient, err := routev1client.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	o.getRoute = func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error) {
+		return routeClient.Routes(namespace).Get(ctx, name, opts)
+	}
```

Comment:
Seems like we only need `BearerToken` from the `RESTConfig` so we may as well only save that and not the full config. 

But looking at how `getRoute` and the token is actually used, I think we can build an alert-fetching closure function here, like this:

```go
routeGeter := func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error) {
		return routeClient.Routes(namespace).Get(ctx, name, opts)
}
o.getAlerts = func(ctx context.Context) ([]byte, error) {
  return inspectalerts.GetAlerts(ctx, routeGetter, cfg.BearerToken
}
```

and then the callsite in `Run()` can just call it easily with just `ctx`, the internal details like routeGetter and token may not be passed around